### DISCUSSION TITLE
[Common Question] How do I know if it's working or if it's a bug?

### DISCUSSION THREAD

**Discussion by pinuke** [brianpetro/obsidian-smart-connections#72] [Q&A] [answered]
"How can I be sure that embeddings.json is being generated correctly. I ask this question because I have a library of 500 notes. After I run the plugin, it keeps showing making smart connections... I try to ask it, but it doesn't seem to be able to read the content of my notes. So how do I know if it's working or if it's a bug?" - [@isaac3203](https://github.com/brianpetro/obsidian-smart-connections/issues/71#issue-1634166769)

Another question that I have is how can I get more details, if there is an error while embedding or using the plugin?

**Comment by pinuke** [comment, marked as answer]
F12 or Ctrl+Shift+I > Console > Check for errors.

Specifically, check for error code 429. If you are hitting your rate limit, see https://github.com/brianpetro/obsidian-smart-connections/discussions/67

If you are getting any errors or having any issues at all, make sure to follow the solution for https://github.com/brianpetro/obsidian-smart-connections/discussions/66#discussioncomment-5374261

If you know of a particularly, problematic file, you can get it's token length using [OpenAI's tool for checking it](https://platform.openai.com/tokenizer)

**Reply by isaac3203** [reply]
Thanks, when I was using my mac today, the notes were processed smoothly and I could visually see them being loaded and processed.
But when I try to use the search box, nothing seems to work.

**Reply by pinuke** [reply]
Thanks for contributing and reporting the problem!

OK, so I would go back to issues and start a new one.

To get the contributors to troubleshoot, get screenshots of:
- failed file list in settings
- the chat not working
- pictures of any errors in devtool console

**Comment by isaac3203** [comment]
<img width="601" alt="image" src="https://user-images.githubusercontent.com/84989111/226841707-bc03aa41-4ed4-4977-9462-67e5943d4f96.png">

**Reply by isaac3203** [reply]
For now I don't have a failure file

**Reply by brianpetro** [reply]
This was helpful. I'm pushing an update now.

**Reply by pinuke** [reply]
@isaac3203 It's nice to see an active community member!

Remember Github discussions is for Q+A. All of your posts here are appearing as an answer to the original question. You're more than welcome to continue to create threads here, but it may confuse new users later in the future.

I recommend keeping bugs and FR's in Github Issues and Q+A's in Github Discussions. Thank you!

**Comment by isaac3203** [comment]
<img width="1508" alt="image" src="https://user-images.githubusercontent.com/84989111/226842772-d0567099-13a7-4379-b69d-00d6f8c53c71.png">

**Reply by isaac3203** [reply]
I typed into the search box on the right but got no concrete response

**Reply by brianpetro** [reply]
This should be fixed with version `1.3.17`.

Let me know if that works for you. 

Thanks for the report,
Brian🌴

**Reply by isaac3203** [reply]
Thank you very much, Mr. Author Brian ! At present, the problems I reported have been fixed and can be used normally! Salute to your dedication and wisdom.
Next, I would like to continue to mention two. question.
First, is the result displayed when using the search the best match for the keywords I searched for? Whether it is irrelevant to the current note content.
Second, is there any way to detect that the plugin is functioning properly when I'm using the chat window, since I've tested it and I can't tell by myself that it's answering with reference to my notes.

**Reply by brianpetro** [reply]
> is there any way to detect that the plugin is functioning properly when I'm using the chat window, since I've tested it and I can't tell by myself that it's answering with reference to my notes.

An update coming very soon will allow you to see the context used for the response.

**Reply by isaac3203** [reply]
I will look forward to it

**Reply by brianpetro** [reply]
@isaac3203 version `1.4.0` adds two buttons to responses from the AI. Both copy context to the clipboard. One is the exact context used by the prompt. The other is the context in the dynamic code block format.

**Reply by isaac3203** [reply]
Thank you very much, I am testing this brand new plugin from the perspective of a user, he is really interesting. I'm still using it, and I hope to give you some feedback after I finish testing, and I can get your answers.

**Reply by isaac3203** [reply]
If possible, I hope to make a video of my testing process to introduce to the audience some details and ideas of using this awesome plugin. Of course, because my English is limited, the test is conducted in Chinese. I don’t know Do you allow.

**Reply by brianpetro** [reply]
It would be great to see a video, regardless of the language, thank you!

**Reply by isaac3203** [reply]
You are too humble. Every ober is lucky to have a talented and diligent developer like you. Thank you for your efforts.

**Comment by isaac3203** [comment]
I will report back a few small problems found during use. First, the network may be unstable during use, resulting in the inability to generate answers. It is manifested as the ellipsis on the answer page keeps beating. I had two lagging experiences, and the first time I got an answer after three minutes. After waiting for about an hour for the second time, I still did not successfully generate an answer. This may have something to do with me using a VPN while abroad. but let me tell you
Second, asking questions in different languages seems to have an impact on plugins. For example, I can't use "I" in Chinese to wake up the plug-in, but I can call the plug-in function by adding the English letter "i" at the beginning of the line. But after my simple six-question comparison experiment. I don't have a complete confirmation of the actual effect of asking questions in English versus other languages. If it is possible, I hope to get your technical answer support, for example, related to its realization principle. An example of my comparison experiment of six questions: I used the English letter "i" as the beginning and Chinese as the main question for the second time. The plug-in successfully implemented calls to 4 of my personal notes, which is the best performance. But when I use English or use pure Chinese to make calls in the future, the performance of the plug-in is not stable.
Third, I would like to make an amendment. As you know, you gave the plugin a total of three copy buttons in the last version. But if I switch to another note and paste it, I will return to the chat page. My scrollbar will go back to the top of the page. This means that I need to scroll down the page again to reach the position I just copied, which will destroy my experience. Perhaps you could consider adding a switch page after the modification to keep the original position. Of course, this is just my personal thoughts and friendly suggestions.
Fourth, your first two rapid version iterations have basically realized everything, my previous vision for our software. I really like this software and I'd be more than happy to recommend it to other ob users. Your work is very meaningful and helps me a lot!

This picture shows the situation of Caton in use.
<img width="680" alt="image" src="https://user-images.githubusercontent.com/84989111/227269355-dcb66b89-7f8b-4e61-9920-74285d5d3bb2.png">


This picture shows the situation when I return to the chat page after switching pages. The key difference is the position of the scroll bar on the right.
<img width="688" alt="image" src="https://user-images.githubusercontent.com/84989111/227269980-04befa4b-ba35-4487-ada9-24f1e2e1bfe5.png">

**Reply by brianpetro** [reply]
The GPT-4 API seemed extra slow for me earlier, including returning some 429 errors. I added some error handling, but it may miss scenarios where the API never returns a response, which happens occasionally.

Regarding self-referential pronouns in other languages to trigger the note-lookup, please see #64 .

Good to note the scrolling. I will keep an eye out for that in my use and patch it as soon as possible.

Thanks for your helpful feedback,
Brian🌴
